sips/sip-16/sip-016-token-metadata.md

```diff
@@ -0,0 +1,394 @@
+
+Appendix A contains a list of trait functions that must meet the following requirements for the return value. The appendix can be extended without changing the ratification status of this SIP.
+
+## Return Value of Metadata URI Functions
+
+The return value must be a `some` value if and only if the metadata reference an existing token, otherwise the value must be `none`. Appendix A specifies the exact meaning of "existing" for each function.
+
+For existing tokens, the inner value of the return value must be a string representing a resolvable URI.
+
+The schema of the resolvable URI is not specified and should be a well-known schema like `https`, `ar`, `ipfs`, `sia`. A `data` URI is also valid, however, the length is limited by this SIP.
+
+If a metadata uri function expects a parameter of type `uint` that identifies a token and the resulting strings contains `{id}`, then the `{id}` part must be replaced by the identifier in decimal format given in the function call.
+
+The resolved data of the URI must be a JSON blob.
+
+## JSON scheme of Metadata
+
+The JSON blob resolved through the uri must follow the following JSON schema.
+
+If metadata were retrieved by a function call containing a token identifier and the string `{id}` exists in any JSON value, it MUST be replaced with the actual token id in decimal format, by all client software that follows this standard.
+
+```
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Token Metadata",
+    "type": "object",
+    "required": ["sip", "name"],
+    "properties": {
+        "sip": {
+            "type": number,
+            "description": "SIP number that defines the JSON schema for metadata. For this SIP, the sip number must be `16`."
+        },
+        "name": {
+            "type": "string",
+            "description": "Identifies the asset which this token represents"
+        },
+        "description": {
+            "type": "string",
+            "description": "Describes the asset which this token represents"
+        },
+        "image": {
+            "type": "string",
+            "description": "A URI pointing to a resource with mime type image/* representing the asset to which this token represents. Consider making any images at a width between 320 and 1080 pixels and aspect ratio between 1.91:1 and 4:5 inclusive. If the token represents a media file of different mime-type or of higher quality defined in property 'raw_media_file_uri', then this image should be used as preview image like a cover for music, or an low-res image."
+        },
+        "attributes": {
+            "type": "array",
+            "description": "Arbitrary attributes. Values may be strings, numbers, object or arrays."
+            "items: {
+                "type": "object",
+                "required": ["value"],
+                "properties": {
```

Comment:
If this field is meant to contain arbitrary attributes, then why are items in this array required to be objects that contain special type restrictions for `display_name` and `trait_name`?

Reply:
Ah, I see this is answered below.  But nevertheless, maybe a comment on why these two fields exist would be appropriate for the `"description"` field, to minimize reader confusion on the first pass?